### PR TITLE
custom user

### PR DESCRIPTION
**Pull request by byteturner**
custom user model added
username deleted
email is auth field
profile relate to user by OneToOne

### CODE REVIEW

File: kinotab_back/profiles/models.py [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 
+from .managers import CustomUserManager
+
 
 class User(AbstractUser):
     """User model use email instead username"""
-
+    username = None
```
Comment (uliantsev-a) [inline kinotab_back/profiles/models.py:10]: It don't delet this field. I think we should remove excess fields.
`CustomUserManager` has addition fields, but main User model still has `first_name` and `last_name`. May be we should use `AbstractBaseUser` or other decision.

image

Comment (byteturner) [reply kinotab_back/profiles/models.py:10]: AbstractUser changed to AbstractBaseUser
Checked by Pylama and cleaned

File: kinotab_back/profiles/signals.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from .models import Profile
+
+
+@receiver(post_save, sender=User)
+def create_user_profile(sender, instance, created, **kwargs):
```
Comment (uliantsev-a) [inline kinotab_back/profiles/signals.py:9]: It's seems thast kwargs not used evereywere. Use **_ instead **kwargs
```suggestion
def create_user_profile(sender, instance, created, **_):
```

File: kinotab_back/profiles/signals.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    if created:
+        Profile.objects.create(user=instance)
+
+
+@receiver(post_save, sender=User)
+def save_user_profile(sender, instance, **kwargs):
```
Comment (uliantsev-a) [inline kinotab_back/profiles/signals.py:15]:
```suggestion
def save_user_profile(sender, instance, **_):
```

File: kinotab_back/profiles/views.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ class UserViewSet(
 
     lookup_field = 'username'
 
-    def create(self, request):
+    def create(self, request, **kwargs):
```
Comment (uliantsev-a) [inline kinotab_back/profiles/views.py:29]:
```suggestion
    def create(self, request, **_):
```

**Pull request merged**